webapp/channels/src/i18n/en.json

```diff
@@ -5184,6 +5184,8 @@
   "user_profile.send.dm.yourself": "Send yourself a message",
   "user.settings.advance.confirmDeactivateAccountTitle": "Confirm Deactivation",
   "user.settings.advance.confirmDeactivateDesc": "Are you sure you want to deactivate your account? This can only be reversed by your System Administrator.",
+  "user.settings.advance.dataPrefetch.Desc": "When disabled, messages and user information will be fetched on each channel load instead of being pre-fetched on startup. This setting is recommended for users with a high unread channel count in order to improve application performance.",
```

Comment:
Is there a way to avoid involving endusers in the decision process and thus avoid the new user setting altogether? Is it feasible to envision making this setting the default?

Reply:
I think it needs to be a conscious decision by the end user to disable prefetch. Prefetching does benefit majority of users and turning it off is of no benefit to them. This will also be hidden behind the `ServiceSettings.EnableClientPerformanceDebugging` as well, we are just hoping to test this on community first and then we can make a decision as to how we want to deliver this to our customers. 

We may end up doing something where if you have more than X number of channels in the pre fetch queue we will disable prefetch on this page load.

Reply:
My chief concern is adding a new user setting, as it seems like something we'd want to do only in exceptional circumstances.

> We may end up doing something where if you have more than X number of channels in the pre fetch queue we will disable prefetch on this page load.

Could we prefetch only up to those X channels for all users? (Prioritizing unread channels in the queue?)

Reply:
I agree that eventually we should remove the user setting and do this automatically. At this stage, it is an experiment to see how much of a benefit it gives to users with a lot of channels.

My gut feeling is that prefetching does not benefit at all - in any cases. And this user setting is a great way to validate that. Once it's in community, we can ask all users to turn it on and see how the reports come in, and whether there's a clear split between users with low channels and high channels and whether there's an inflection point if any.

So let's go with a user setting now. Collect data as much as possible, and then come up with an automatic solution.

Reply:
I agree with @agnivade, we can use this experimentally on community and come up with a plan to make it automatic afterwards